### DISCUSSION TITLE
0.4.1-stable has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#489] [Announcements]
_This is a quick patch release to fix an issue with `ConcavePolygonShape3D`. Be sure to also read the release notes for [**0.4.0-stable**](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.4.0-stable) if you're upgrading from 0.3.0-stable._

This release supports [**Godot 4.1**](https://github.com/godotengine/godot/releases/tag/4.1-stable).

## Changelog

These are the notable changes that have been made since [0.4.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.4.0-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.4.0-stable...v0.4.1-stable).

Any breaking changes are denoted with ⚠️.

### Fixed

- Fixed issue where colliding with certain types of degenerate triangles in `ConcavePolygonShape3D` would cause the application to hang or emit a vast amount of errors.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.4.1-stable'>0.4.1-stable</a>.</em>

**Comment by mgiraud** [comment]
If you are interested, here's the pinball prototype for windows using the Generic6DofJoint (from 0.4.0) with a spring.
Shape margins are disabled, Physics FPS = 240 💨. Everything is scaled *10, meaning the ball is 30cm diameter 😆 
https://eowindil.itch.io/pinball-protoype?secret=n7sPOeBavx0LhG9Bz7QCx5a3C0

**Reply by mihe** [reply]
Hm, when I click "Download Now" I just get a browser message box saying "invalid game" and nothing gets downloaded.

**Reply by mgiraud** [reply]
Sorry, new link https://eowindil.itch.io/pinball-protoype, password is "pinball"

**Reply by mihe** [reply]
That's pretty neat! Takes me back to playing that [Space Cadet](https://en.wikipedia.org/wiki/Full_Tilt!_Pinball#3D_Pinball_for_Windows_%E2%80%93_Space_Cadet) pinball game that was bundled with Windows ME/XP.

I'm curious, how did you end up rotating the flippers in the end? If you're using 6DOF springs you now also have the option of using the "Equilibrium Point" properties on each axis, which is basically just an offset that controls where the spring is at rest. It might be worth considering if you're not using it already.

**Reply by mgiraud** [reply]
Yes i've set a small offset with the equilibrium point which kind of stabilize the spring in the lower position. The flippers do not have angular limits but are restrained by invisible colliders. This allows for a small bounce when the ball hits the flipper
![image](https://github.com/godot-jolt/godot-jolt/assets/11818237/243c37c4-ff69-4981-b037-66d68619f43f)
The "kick" uses the angular motor

**Reply by mihe** [reply]
That's quite the setup. I suppose once the custom joint nodes are in you might be able to replace those invisible bodies with soft limits instead.
